### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by Giost**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] PR has a descriptive title
- [x] PR contains closely related changes only (should not have many unrelated changes)
- [x] STEP pod members have been added as reviewers
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
- Setup of the maven project.
- Add a sample web page, servlet and servlet test.
- Setup automatic building and testing on GitHub when there is a PR.

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [x] Other... Please describe: Project setup and configuration

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
```
Comment (dola) [inline .github/workflows/maven.yml:4]: suggestion by Jacek: rename this to make it clearer that this also runs tests

File: src/test/java/com/google/sps/servlets/SampleServletTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+@RunWith(JUnit4.class)
+public final class SampleServletTest {
+
+  @Test
+  public void testSampleServletResponse() throws IOException {
```
Comment (dola) [inline src/test/java/com/google/sps/servlets/SampleServletTest.java:32]: nice!

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  <build>
+    <plugins>
+      <!-- Provides `mvn package appengine:run` for local testing
+           and `mvn package appengine:deploy` for deploying. -->
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.2.0</version>
+        <configuration>
+          <deploy.projectId><!-- TODO: insert project id --></deploy.projectId>
```
Comment (zlorf) [inline pom.xml:50]: You can insert project id now. :)

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          <deploy.version>1</deploy.version>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+</project>
```
Comment (zlorf) [inline pom.xml:56]: Please make sure that each file has a newline at the end. Best option would be for you to configure IDE to automatically ensure whitespaces are correct (newline at the end + no trailing whitespace) when file is saved.

**Pull request merged**